docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,129 @@ The names of the functions are also available as the keys of the
 building-hacks
 ==============
 
-This plugin overwrites some methods in workshop df class so that mechanical workshops are possible. Although
-plugin export a function it's recommended to use lua decorated function.
+This plugin overwrites some methods in workshop df class to change and extend workshop
+functionality. See exported function list for details.
 
 .. contents::
   :local:
 
 Functions
 ---------
 
-``registerBuilding(table)`` where table must contain name, as a workshop raw name, the rest are optional:
+* ``setOwnableBuilding(workshop_type, enable)``
 
-    :name:
-        custom workshop id e.g. ``SOAPMAKER``
+  Set workshop to be included in zones (such as bedroom or inn).
 
-        .. note:: this is the only mandatory field.
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :enable:   boolean value to enable or disable this functionality
 
-    :fix_impassible:
-        if true make impassable tiles impassable to liquids too
-    :consume:
-        how much machine power is needed to work.
-        Disables reactions if not supplied enough and ``needs_power==1``
-    :produce:
-        how much machine power is produced.
-    :needs_power:
-        if produced in network < consumed stop working, default true
-    :gears:
-        a table or ``{x=?,y=?}`` of connection points for machines.
-    :action:
-        a table of number (how much ticks to skip) and a function which
-        gets called on shop update
-    :animate:
-        a table of frames which can be a table of:
+* ``fixImpassible(workshop_type, enable)``
 
-        a. tables of 4 numbers ``{tile,fore,back,bright}`` OR
-        b. empty table (tile not modified) OR
-        c. ``{x=<number> y=<number> + 4 numbers like in first case}``,
-           this generates full frame useful for animations that change little (1-2 tiles)
+  Set workshop non walkable tiles to also block liquids (i.e. water and magma).
 
-    :canBeRoomSubset:
-        a flag if this building can be counted in room. 1 means it can, 0 means it can't and -1 default building behaviour
-    :auto_gears:
-        a flag that automatically fills up gears and animations. It looks over the building definition for gear icons and maps them.
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :enable:   boolean value to enable or disable this functionality
```

Comment:
should this be enabled per workshop *type* or per workshop type and *state*? If not, and it's just by type, then does it make sense to dynamically toggle? Similar question to setOwnableBuilding above.

Reply:
ditto. Will remove the enable